[bryonk] iP

bryonk opened the pull request:
### Chat Chatbot

> "Insert quote here" - Me

Chat frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/bryonk/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [ ] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

bryonk committed:
Level 0. Rename, Greet, Exit

bryonk committed:
Level 1. Echo

bryonk committed:
Level 2. Add, List

bryonk committed:
Level 3. Mark as Done

bryonk committed:
Level 4. ToDos, Events, Deadlines

bryonk committed:
Automated Text UI Testing

bryonk committed:
rename Duke to Chat

bryonk committed:
Level 5. Handle Errors

bryonk committed:
Level 6. Delete and A-Enums: Use Enums

bryonk committed:
Add TaskList to abstract list of tasks and Storage for save to file functionality

bryonk committed:
Merge branch 'branch-Level-7'

bryonk committed:
Update Deadline and Event to support DateTime objects

bryonk committed:
Merge branch 'branch-Level-8'

bryonk committed:
Move and update files to respective packages

bryonk committed:
Merge branch 'add-gradle-support'

bryonk committed:
Add JUnit tests for Parser and TaskList

bryonk committed:
Add JavaDocs into files

bryonk committed:
Update coding standards in files

bryonk committed:
Add find function to search for tasks

bryonk committed:
Merge branch 'branch-A-JavaDoc'

bryonk committed:
Merge branch 'branch-A-CodingStandard'

bryonk committed:
Merge branch 'branch-Level-9'

leedeen01 commented on src/main/java/chat/exceptions/ChatEditException.java:
```diff
@@ -0,0 +1,7 @@
+package chat.exceptions;
+
+public class ChatEditException extends ChatException {
```
I like how you have proper naming

leedeen01 commented on src/main/java/chat/parser/Function.java:
```diff
@@ -0,0 +1,8 @@
+/**
+ * Contains the different functions accepted by the program.
+ */
+public enum Function {
```
I like how you used enum to limit the different functions accepted by the program

leedeen01 commented on src/main/java/chat/tasklist/TaskList.java:
```diff
@@ -0,0 +1,135 @@
+        }
+    }
+
+    public int getSize() {
```
Perhaps javadocs can be added to improve code clarity

leedeen01 commented on src/main/java/chat/parser/Parser.java:
```diff
@@ -0,0 +1,61 @@
+     * @return Job object containing the function and description if present.
+     * @throws ChatParseException If the function is not recognised.
+     */
+    public static Job parseInput(String input) throws ChatParseException {
```
Perhaps using Varargs will allow you to pass a variable number of arguments to a method

leedeen01 reviewed (commented):
LGTM. Just some nits to fix.

Fieash commented on src/main/java/Event.java:
```diff
@@ -1,20 +1,33 @@
+            this.to = LocalDateTime.from(d.parse(to.trim()));
+        } catch (DateTimeException e) {
+            throw new ChatAddException("ChatAddException: Date time in wrong format (dd/MM/yyyy HHmm)");
+        }
```
should `DateTimeFormatter` be abbreviated as `d` ? otherwise, clear and easy to understand code!

Fieash commented on src/main/java/Deadline.java:
```diff
@@ -1,18 +1,28 @@
     public Deadline(String description, String by) {
         super(description);
-        this.by = by;
+        try {
```
similar to the other comment, should `DateTimeFormatter` here be abbreviated as `d`?

Fieash reviewed (commented):
Looks good!
some cosmetic changes requested

bryonk committed:
Update javadoc in TaskList.java

bryonk committed:
Update code to meet checkstyle requirements

bryonk committed:
Add checkstyle configs

bryonk committed:
Add GUI feature

bryonk committed:
Merge branch 'branch-A-CheckStyle'

bryonk committed:
Merge branch 'branch-Level-10'

bryonk committed:
Add assertions to code

There are currently no assertions to document assumptions.

Addition of assertions will ensure that the input is as required.

Let's add assertions to classes to ensure that the assumptions
are correct.

bryonk committed:
Update files to improve code quality

The code quality in some classes can be improved to fix readability.

Improving code quality will help users read and understand the
code better.

Let's,
* follow the Single Level of Abstraction Principle
* convert magic Strings to named constants
* make the happy path prominent.

bryonk committed:
Merge pull request #1 from bryonk/branch-A-Assertions

Add assertions to code

bryonk committed:
Merge branch 'master' into branch-A-CodeQuality

bryonk committed:
Merge pull request #2 from bryonk/branch-A-CodeQuality

Update files to improve code quality

bryonk committed:
Add feature to detect for duplicate task

bryonk committed:
Merge branch 'branch-B-DetectDuplicate'

bryonk committed:
Update files to improve code quality and user images

bryonk committed:
Update user guide
